Enhance command to handle contenteditable elements

stevedorries opened the pull request:
This PR adds the ability to fill elements that have the contenteditable attribute set. This is needed in general for contenteditable elements and in my specific use case to update the value of FROALA editor instances.

stevedorries committed:
Enhance command to handle contenteditable elements

DanielFerrariR commented:
Thanks for your PR.

Some thoughts about this idea:

1 - Maybe would be nice to extend this overwrite option to be used in the inputs and area inputs too.
2 - I think the default behavior should be overwrite everything. Is more common to use fill command to fill/replace the content of an input than to keep and increase it, at least in my used cases.

Please fix the failing tests and let me know what you think.

stevedorries committed:
Changes default behavior to overwrite
Adds support for options to input and textarea elements

stevedorries commented:
> Thanks for your PR.
> 
> Some thoughts about this idea:
> 
> 1 - Maybe would be nice to extend this overwrite option to be used in the inputs and area inputs too.
> 2 - I think the default behavior should be overwrite everything. Is more common to use fill command to fill/replace the content of an input than to keep and increase it, at least in my used cases.
> 
> Please fix the failing tests and let me know what you think.

Probably a good idea to have all elements that can use the fill command also obey the potential options, extending it thusly then forces overwrite to be the default since doing otherwise would be a breaking change for existing code. 

I extended the options to apply to input and textarea, reversed the default behavior, and added a prepend option that defaults to false. Also, I added test cases for the new option.

DanielFerrariR commented:
Thanks!

@all-contributors please add @stevedorries for ideas, tests and code
